org.codeberg.dnkl.foot.desktop

```diff
@@ -0,0 +1,10 @@
+[Desktop Entry]
```

Comment:
This file should be submitted and maintained upstream.

Reply:
Sorry for the noob question, but what does it mean "maintained upstream"?

Reply:
It means that it should be put in the Codebrg repo, not here in the Flathub repo

Reply:
He has no affiliation with the domain. The manifest is blindly spawning a terminal outside the sandbox (this seems to be for packaging only). This manifest should never reach flathub as it doesnt use flatpak internals in any sense other than packaging .

Please see responses from upstream: https://codeberg.org/dnkl/foot/issues/1129#issuecomment-598905

Reply:
A Linux desktop appication without a .desktop, it is a bug. Upstream is the original developers, they should be the one maintaining both the .desktop, icon and appdata (neither is flatpak specific) and we try to make this a policy.

Reply:
> He has no affiliation with the domain.

Yes I wanted to package it on my own (maybe if there was someone in whom it would spark an interest in flatpak, there is a `help wanted` label on the [issue](https://codeberg.org/dnkl/foot/issues/1129) after all.

>The manifest is blindly spawning a terminal outside the sandbox (this seems to be for packaging only). This manifest should never reach flathub as it doesnt use flatpak internals in any sense other than packaging .

It doesn't spawn a terminal outside the sandbox. It just spawns a shell on the host, using [host-spawn](https://github.com/1player/host-spawn) because it solves some things with the pty that `flatpak-spawn --host` does not.
I don't know what's wrong on spawning the host shell, [BlackBox](https://flathub.org/apps/details/com.raggesilver.BlackBox) does it too [here](https://gitlab.gnome.org/raggesilver/blackbox/-/blob/main/src/widgets/Terminal.vala#L302).

As for the "maitained upstream" issue, there is a [foot.desktop](https://codeberg.org/dnkl/foot/src/branch/master/foot.desktop) in the repo, but there is no appstream and the main dev has said, [this](https://codeberg.org/dnkl/foot/issues/1129#issuecomment-597445):

> FWIW, I really really really do not want to add flatpak specific code to foot. If that means foot cannot be packaged as a flatpak, then so be it.

So I tried to overcome it, without changing the code, with just rewriting the `$SHELL` var and prepend `host-spawn`. 

I could try to convince the author to add appstream file to the repo. After all it's not flatpak specific.

Reply:
appstream file ARE NOT FLATPAK SPECIFIC.

Only the manifest is.

Reply:
> The manifest is blindly spawning a terminal outside the sandbox (this seems to be for packaging only). This manifest should never reach flathub as it doesnt use flatpak internals in any sense other than packaging .

To clarify, there's no other way to ship a terminal with flatpak. The shell inside the sandbox isn't exactly as useful as the one in the host, and that's what multiple projects already do.

Reply:
> To clarify, there's no other way to ship a terminal with flatpak. The shell inside the sandbox isn't exactly as **useful** as the one in the host, and that's what multiple projects already do.

There are ways to ship the terminal. but as you say, "useful" is the keyword here. And that is kind of my point. The feature request on upstream was created by @micro8 with the argument of **wanting sandboxing features**. Totally bypassing upstream on the very issue one created to begin with, then submitting pr claiming critierias are met. The problem isnt that the application is on flathub or how one end up with defining the manifest. -Going rogue is. Hope that clarifies.